pkg/sources/source_manager.go

```diff
@@ -0,0 +1,168 @@
+package sources
+
+import (
+	"fmt"
+	"runtime"
+	"sync"
+	"sync/atomic"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/sourcespb"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+// handle uniquely identifies a Source given to the manager to manage. If the
+// SourceManager is connected to the API, it will be equivalent to the unique
+// source ID, otherwise it behaves as a counter.
+type handle int64
+
+type SourceManager struct {
+	api apiClient
+	// Map of handle to source information.
+	handles     map[handle]sourceTemplate
+	handlesLock sync.Mutex
+	// Counters for assigning handle and job IDs for headless managers.
+	headlessHandleCounter int64
+	headlessJobCounter    int64
+	// Pool limiting the amount of concurrent sources running.
+	pool errgroup.Group
+	// Downstream chunks channel to be scanned.
+	outputChunks chan *Chunk
+}
+
+// sourceTemplate is an aggregate struct to hold the Source and its name.
+type sourceTemplate struct {
+	name   string
```

Comment:
Are you expecting to use `name` in logs and the like? I don't see it used anywhere else (but I might just be missing it).